pkg/pool-weighted/contracts/WeightedMath.sol

```diff
@@ -192,7 +192,49 @@ library WeightedMath {
         );
 
         uint256 bptOut = (invariantRatio > FixedPoint.ONE)
-            ? bptTotalSupply.mulDown(invariantRatio.sub(FixedPoint.ONE))
+            ? bptTotalSupply.mulDown(invariantRatio - FixedPoint.ONE)
+            : 0;
+        return bptOut;
+    }
+
+    function _calcBptOutGivenExactTokenIn(
+        uint256 balance,
+        uint256 normalizedWeight,
+        uint256 amountIn,
+        uint256 bptTotalSupply,
+        uint256 swapFeePercentage
+    ) internal pure returns (uint256) {
+        // BPT out, so we round down overall.
+
+        uint256 amountInWithoutFee;
+        {
+            uint256 balanceRatioWithFee = balance.add(amountIn).divDown(balance);
+            uint256 invariantRatioWithFees = balanceRatioWithFee.mulDown(normalizedWeight).add(
+                normalizedWeight.complement()
```

Comment:
```suggestion
            // This assumes that the sum of all weights equals FixedPoint.ONE, which may not be the case during a
            // gradual weight change due to interpolation rounding error. This will result in a small difference between the
            // output of this function and the equivalent `_calcBptOutGivenExactTokensIn` call.
            uint256 invariantRatioWithFees = balanceRatioWithFee.mulDown(normalizedWeight).add(
                normalizedWeight.complement()
```

Reply:
It'd be good to better explain *what* this difference will be, once we understand it.

Reply:
This also occurs in the situation where we have no weight change due to normalization of the stored weights.

Reply:
I was already thinking of the no normalization case :upside_down_face: